cpmpy/expressions/globalfunctions.py

```diff
@@ -362,4 +362,60 @@ def get_bounds(self):
         """
         Returns the bounds of the (numerical) global constraint
         """
-        return 1, len(self.args)
\ No newline at end of file
+        return 1, len(self.args)
+
+
+class NValueExcept(GlobalFunction):
+
+    """
+        The NValueExceptN constraint counts the number of distinct values,
+            not including value N, if any argument is assigned to it.
+    """
+
+    def __init__(self, arr, n):
+        if not is_any_list(arr):
+            raise ValueError("NValueExcept takes an array as input")
+        if not is_num(n):
+            raise ValueError(f"NValueExcept takes an integer as second argument, but got {n} of type {type(n)}")
+        super().__init__("nvalue_except",[arr, n])
+
+    def decompose_comparison(self, cmp_op, cpm_rhs):
+        """
+        NValue(arr) can only be decomposed if it's part of a comparison
+
+        Based on "simple decomposition" from:
+            Bessiere, Christian, et al. "Decomposition of the NValue constraint."
+            International Conference on Principles and Practice of Constraint Programming.
+            Berlin, Heidelberg: Springer Berlin Heidelberg, 2010.
+        """
+        from .python_builtins import sum, any
+
+        arr, n = self.args
+        arr = cpm_array(arr)
+        lbs, ubs = get_bounds(arr)
+        lb, ub = min(lbs), max(ubs)
+
+        constraints = []
+
+        # introduce boolvar for each possible value
+        bvars = boolvar(shape=(ub + 1 - lb))
+        idx_of_n = n - lb
+        if idx_of_n >= 0:
```

Comment:
and n < ub? to avoid index out of bound if big n is given

Reply:
Ah yes indeed